File: subject-service/src/main/java/kz/baltabayev/subjectservice/controller/advice/GlobalExceptionHandler.java
package kz.baltabayev.subjectservice.controller.advice;

import kz.baltabayev.subjectservice.exception.SubjectNotFoundException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ProblemDetail;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(SubjectNotFoundException.class)
    ProblemDetail handlerSubjectNotFoundException(SubjectNotFoundException exception) {
        return ProblemDetail.forStatusAndDetail(HttpStatus.NOT_FOUND, exception.getMessage());
    }
}

File: faculty-department-service/src/main/kotlin/kz/baltabayev/facultydepartmentservice/controller/advice/GlobalExceptionHandler.kt
package kz.baltabayev.facultydepartmentservice.controller.advice

import jakarta.persistence.EntityNotFoundException
import org.springframework.http.HttpStatus
import org.springframework.http.ProblemDetail
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(EntityNotFoundException::class)
    fun handleEntityNotFoundException(e: EntityNotFoundException): ResponseEntity<ProblemDetail> {
        val problemDetail = ProblemDetail.forStatusAndDetail(HttpStatus.NOT_FOUND, e.localizedMessage)

        return ResponseEntity.status(HttpStatus.NOT_FOUND)
            .body(problemDetail)
    }
}